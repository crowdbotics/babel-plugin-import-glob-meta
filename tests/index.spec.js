const babel = require('babel-core');
const plugin = require('../');
const path = require('path');
const fs = require('fs');

function loadFixture(name) {
  const fixture = path.join(__dirname, "fixtures", name);
  return {
    content: fs.readFileSync(fixture, 'utf8'),
    filename: fixture
  }
}

it('converts default glob imports', () => {
  const { content, filename } = loadFixture("index.js");
  const { code } = babel.transform(content, { plugins: [plugin], filename });
  expect(code).toMatchSnapshot();
});

it('throws error on named glob imports', () => {
  const { content, filename } = loadFixture("index.bad.js");
  expect(() => babel.transform(content, { plugins: [plugin], filename }))
    .toThrow('Can only import the default export from a glob pattern');
});
