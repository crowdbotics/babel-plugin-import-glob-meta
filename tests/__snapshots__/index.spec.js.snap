// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converts default glob imports 1`] = `
"import _ig from \\"./module-B/index.js\\";
import _ig2 from \\"./moduleA/index.js\\";
let modules = [{
  \\"name\\": \\"moduleB\\",
  \\"value\\": _ig,
  \\"path\\": \\"./module-B/index.js\\",
  \\"package\\": \\"@modules/moduleB\\"
}, {
  \\"name\\": \\"moduleA\\",
  \\"value\\": _ig2,
  \\"path\\": \\"./moduleA/index.js\\",
  \\"package\\": \\"@modules/moduleA\\"
}];
import _ig3 from \\"./module-B/options.js\\";
import _ig4 from \\"./moduleA/options.js\\";
let options = [{
  \\"name\\": \\"moduleB\\",
  \\"value\\": _ig3,
  \\"path\\": \\"./module-B/options.js\\",
  \\"package\\": \\"@modules/moduleB\\"
}, {
  \\"name\\": \\"moduleA\\",
  \\"value\\": _ig4,
  \\"path\\": \\"./moduleA/options.js\\",
  \\"package\\": \\"@modules/moduleA\\"
}];"
`;
